### PR TITLE
feat: add scalar popover

### PR DESCRIPTION
**Pull request by hwkr**
Little followup PR to #1557 to add popovers and extend scalar floating a little bit for some future usage I foresee with middlewares.

Also I forgot to add the dropdown and floating components to the components export...

https://github.com/scalar/scalar/assets/6374090/18147245-1f6e-4c0f-88e4-12118e6ec8de

### CODE REVIEW

**Review by amritk — APPROVED** [review]
Looks good, nice to have this in a component so we can replace with css anchor when its available!

**Review by marclave — APPROVED** [review]
nice work @hwkr ! @amritk left a slight comment but can be merged after addressed ✨

File: packages/components/src/components/ScalarFloating/ScalarFloating.vue [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+} from '@floating-ui/vue'
 import { type Ref, computed, ref } from 'vue'
 
 import type { FloatingOptions } from './types'
 import { useResizeWithTarget } from './useResizeWithTarget'
 
 const props = defineProps<FloatingOptions>()
 
+defineSlots<{
+  /** The reference element for the element in the #floating slot */
```
[the remark this answers is not in the document]
Comment (amritk) [reply packages/components/src/components/ScalarFloating/ScalarFloating.vue:19]: Confirmed it still works!

**Comment by amritk** [comment]
@marclave @hwkr comment redacted

**Pull request merged**